examples/cpp/vamana_iterator.cpp

```diff
@@ -68,42 +68,34 @@ void check(bool value, std::string_view expr, svs::lib::detail::LineInfo linfo)
     }
 
     // Build the index.
-    return svs::Vamana::build<float>(build_parameters, std::move(data), svs::DistanceL2{});
+    return svs::index::vamana::auto_build(
+        build_parameters, std::move(data), svs::DistanceL2{}, 1
+    );
 }
 //! [Example Index Construction]
 
-void demonstrate_default_schedule() {
+void demonstrate_iterator() {
     //! [Setup]
     auto index = make_example_index();
 
-    // Base search parameters for the iterator schedule.
-    // This uses a search window size/capacity of 4.
-    auto base_parameters = svs::index::vamana::VamanaSearchParameters{}.buffer_config({4});
-
-    // The default schedule take base parameters and a batch size.
     // Each iteration will yield 3 elements that have not been yielded previously.
     size_t batchsize = 3;
-    auto schedule = svs::index::vamana::DefaultSchedule{base_parameters, batchsize};
 
     // Create a batch iterator over the index for the query.
-    // After the constructor returns, the contents of the first batch will be available.
     auto itr = [&]() {
         // Construct a query a query in a scoped block to demonstrate that the iterator
         // maintains an internal copy.
         auto query = std::vector<float>{3.25, 3.25, 3.25, 3.25};
 
-        // Make a batch iterator for the query using the provided schedule.
-        return index.batch_iterator(svs::lib::as_const_span(query), schedule);
+        // Make a batch iterator for the query.
+        return svs::VamanaIterator(index, svs::lib::as_const_span(query));
```

Comment:
Since the example will not show `index.batch_iterator` anymore, we should mention it somewhere else in the docs (aside from the method doc) because it is still useful and users should be aware of it. This could be a separate PR.